api/server/server.go

```diff
@@ -787,15 +789,19 @@ func (s *Server) postImagesCreate(eng *engine.Engine, version version.Version, w
 		}
 		imageImportConfig.ContainerConfig = newConfig
 
-		opErr = s.daemon.Repositories().Import(src, repo, tag, imageImportConfig)
-	}
+		err = s.daemon.Repositories().Import(src, repo, tag, imageImportConfig)
 
-	if opErr != nil {
+	}
+	if err != nil {
+		if !output.Flushed() {
+			return err
+		}
 		sf := streamformatter.NewStreamFormatter(useJSON)
-		return fmt.Errorf(string(sf.FormatError(opErr)))
+		output.Write(sf.FormatError(err))
```

Comment:
If this method fails, shouldn't still return an error indicating that it failed? Perhaps, the calling code should be formatting the error so this gets fixed for other calls besides `postImagesCreate`.